### ISSUE TITLE
`docker stack deploy` errors with volume labels in compose v3 yaml

### ISSUE THREAD

**Issue by jlhawn** [docker/docker#30297]
**Description**

`docker stack deploy` doesn't like it when you give it a compose yaml file which tries to add labels to a volume:

```yaml
...
volumes:
  vol1:
    driver: local
    labels:
      com.docker.ucp.access.label: test
```

Error: `labels Additional property labels is not allowed`

**Steps to reproduce the issue:**
1. Create a compose file which tries to specify volume labels.
2. Try to deploy the stack
3. See the error that results

**Describe the results you received:**

An error message: `labels Additional property labels is not allowed`

**Describe the results you expected:**

I expected it to deploy the services, networks, and volumes in the stack with no issue.

Docker 1.13.0

**Comment by thaJeztah** [comment]
Orthogonal to this issue, but something to note is that there's no _update_ endpoint for volumes, so editing the labels after the volume is created won't be possible

**Comment by jlhawn** [comment]
Corresponding issue: https://github.com/docker/compose/issues/4359

**Comment by dnephin** [comment]
This is already fixed in master as of #29609. We need to cherry pick that branch anyway for other deploy changes.

**Comment by jlhawn** [comment]
@dnephin It looks like that PR (and `master`) both have the `"labels"` property in the wrong place in the schema for the volume definition. Looks like we'll need to apply the same change that @shin- made here: https://github.com/docker/compose/pull/4360

**Comment by dnephin** [comment]
Yes, sorry I was looking at the wrong thing. I have a PR up that fixes it now: #29883

**Comment by vdemeester** [comment]
Should be fixed by #29883, closing
